prov/rxm/src/rxm.h

```diff
@@ -299,8 +302,18 @@ struct rxm_rndv_hdr {
 	uint8_t count;
 };
 
-#define rxm_pkt_rndv_data(rxm_pkt) \
-	((rxm_pkt)->data + sizeof(struct rxm_rndv_hdr))
+struct rxm_rndv_raw_key_hdr {
+	uint64_t base_addr;
+	uint8_t raw_key[];
+};
+
+#define rxm_rndv_hdr_size(rxm_ep) \
+	(sizeof(struct rxm_rndv_hdr) + \
+	 RXM_IOV_LIMIT * (sizeof(struct rxm_rndv_raw_key_hdr) + \
+	 (rxm_ep)->msg_info->domain_attr->mr_key_size))
+
+#define rxm_pkt_rndv_data(rxm_pkt, rxm_ep) \
+	((rxm_pkt)->data + rxm_rndv_hdr_size(rxm_ep))
```

Comment:
@a-ilango What data is carried as part of the rendezvous packet?  Is this cq data only?

Assuming that this the case _and_ we keep the base rxm_rndv_hdr even if MR_RAW is enabled, the cq data could remain immediately after the base hdr, which would eliminate the need for the rxm_rndv_hdr_size macro.

Reply:
The data refers to the application data (actual message). Rendezvous request packets would carry application data if user had requested FI_BUFFERED_RECV. This is needed to satisfy FI_OPT_BUFFERED_MIN.